cfme/tests/infrastructure/test_infra_object_tag_visibility.py

```diff
@@ -0,0 +1,110 @@
+from cfme.infrastructure.virtual_machines import Vm, Template
+from cfme.web_ui import mixins
+from fixtures.provider import setup_one_or_skip
+from utils.appliance.implementations.ui import navigate_to
+from utils.log import logger
+from utils.providers import ProviderFilter
+
+
+pytestmark = [test_requirements.tag, pytest.mark.tier(2)]
+
+
+@pytest.fixture(scope='module')
+def a_provider(request):
+    prov_filter = ProviderFilter(classes=[VMwareProvider])
+    return setup_one_or_skip(request, filters=[prov_filter])
+
+
+test_items = [
+    ('providers', VMwareProvider),
+    ('clusters', Cluster),
+    ('hosts', Host),
+    ('data_stores', Datastore),
+    ('vms', Vm),
+    ('templates', Template)
+]
+
+
+@pytest.fixture(params=test_items, ids=["{}".format(test_id[0]) for test_id in test_items],
+                scope='function')
+def testing_vis_object(request, a_provider, appliance):
+    """ Fixture creates class object for tag visibility test
+    Returns: class object of certain type
+    """
+    collection_name, param_class = request.param
+    if collection_name == 'clusters':
+        test_items = a_provider.get_clusters()
+    else:
+        test_items = getattr(appliance.rest_api.collections, collection_name)
+
+    if not test_items:
+        pytest.skip('No content found for test!')
+
+    if collection_name == 'data_stores':
+        return param_class(name=test_items[0].name)
+    elif collection_name == 'templates':
+        for index, test_item_value in enumerate(test_items):
+            if test_item_value.vendor == 'vmware':
+                return param_class(name=test_item_value.name, provider=a_provider)
+    elif collection_name != 'providers':
+        test_obj = param_class(name=test_items[0].name, provider=a_provider)
+    else:
+        test_obj = a_provider
+    return test_obj
+
+
+@pytest.fixture(scope='function')
+def check_item_visibility(request, appliance, tag, user_restricted, testing_vis_object):
+    def _check_item_visibility(visibility_result):
+        """
+        Args:
+            visibility_result: pass 'True' is item should be visible,
+                               'False' if not
+        """
+        navigate_to(testing_vis_object, 'Details')
+        if visibility_result:
+            mixins.add_tag(tag)
+        else:
+            mixins.remove_tag(tag)
+        assert visibility_result == _check_if_visible(testing_vis_object)
+
+    def _check_if_visible(visibility_object):
+        """ Checks if user can access item from UI,
+            and deatil page is available for user
+        Args:
+             visibility_object: item object that should be tested for visibility
+        Returns: True if item is visible for user, and False if not
+        """
+        visible = False
+        with user_restricted:
+            try:
+                navigate_to(visibility_object, 'Details')
+                visible = True
+            except:
```

Comment:
**REQUIRED**
Please always specify an exception. At least `Exception` must be caught.